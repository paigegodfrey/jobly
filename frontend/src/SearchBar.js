import React, { useState } from 'react';

function SearchBar({ filterBySearchObject }) {
  const [searchObject, setSearchObject] = useState({ search: "" });

  const handleSubmit = evt => {
    evt.preventDefault();
    console.log('searchObject is', searchObject)

    filterBySearchObject(searchObject);
    console.log('search term post', searchObject)
  };

  const handleChange = evt => {
    const { name, value } = evt.target;
    setSearchObject(sData => ({
      ...sData,
      [name]: value
    }));
  };

  return (
    <div>
      <form onSubmit={handleSubmit}>
        <input
          id="search"
          name="search"
          type="text"
          placeholder="Search.."
          value={searchObject.search}
          onChange={handleChange}
        />
        <button>Submit</button>
      </form>
    </div>
  )
}
export default SearchBar;