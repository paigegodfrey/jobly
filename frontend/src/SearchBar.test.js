import React from "react";
import { render } from "@testing-library/react";
import SearchBar from "./SearchBar";

it("renders without crashing", function() {
  render(<SearchBar />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<SearchBar />);
  expect(asFragment()).toMatchSnapshot();
});
