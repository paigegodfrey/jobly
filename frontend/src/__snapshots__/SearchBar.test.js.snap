// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="SearchBar mb-4"
  >
    <form
      class="form-inline"
    >
      <div
        class="input-group flex-grow-1"
      >
        <input
          class="form-control form-control-lg"
          name="search"
          placeholder="Search.."
          type="text"
          value=""
        />
        <div
          class="input-group-append input-group-btn"
        >
          <button
            class="btn btn-lg btn-primary"
          >
            Submit
          </button>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
